examples/todo/test/acceptance/application.acceptance.ts

```diff
@@ -19,6 +20,21 @@ describe('Application', () => {
   before(givenAnApplication);
   before(async () => {
     await app.boot();
+
+    /**
+     * Override DataSource to not write to file for testing. Since we aren't
+     * persisting data to file and each injection normally instatiates a new
+     * instance, we must change the BindingScope to a singleton so only one
+     * instance is created and used for all injections (preserving access to
+     * the same memory space).
+     */
+    app.bind('datasources.config.db').to({
+      name: 'db',
+      connector: 'memory',
```

Comment:
-1 for duplicating most of the configuration of `db` datasource. Could you please try to find a way how to rewrite this code so that it changes only the relevant flags (e.g. set `file` to `undefined` - I don't remember exact `memory` config options)?

Reply:
Just binding `{file: undefined}`won't work as a `DataSource` needs the `name` and `connector`. See my comment above with regards to merging the bound value with the default config.

Reply:
I guess an alternative would be to do the merge here instead of the datasource -- I think I like this option better.

```
const dsKey = 'datasources.config.db';
const config = await app.get(dsKey);
app.bind(dsKet).to(Object.assign(config, {file: undefined}));
```

Reply:
> I guess an alternative would be to do the merge here instead of the datasource -- I think I like this option better.

Yes please, that's what I had in mind 👍

Reply:
I created a follow-up issue to improve the overall design to allow tests to modify the default datasource configuration, see #1396

For the scope of this pull request, I am proposing to go with a solution that's fastest to land. We can use the code snippet shown in https://github.com/strongloop/loopback-next/pull/1385/files#r192946885, or even leave `{file: undefined}` change out of scope and simply use the same datasource config as the application uses at runtime.

Reply:
Does `datasources.config.db` follow any convention?

Reply:
convention for naming or it's config? It's an optional binding a user can set to override the `DataSource` config (default coming from the generated JSON file). This is more for testing.

Reply:
Does `@loopback/boot` bind datasource config to `datasources.config.<dsName>`?